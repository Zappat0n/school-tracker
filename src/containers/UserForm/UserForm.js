import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { logIn, signUp } from '../../api/queries';
import { addRedirect, save } from '../../slices/userSlice';

const UserForm = (props) => {
  const { action } = props;
  const redirect = useSelector((state) => state.user.redirect);
  const dispatch = useDispatch();

  async function getLogIn(email, password) {
    const response = await logIn(email, password);
    if (response.access_token) {
      dispatch(save({
        email,
        password,
        token: response.access_token,
      }));
      return true;
    }
    return false;
  }

  async function getSignUp(email, password) {
    if (signUp(email, password)) {
      getLogIn(email, password);
    }
  }

  async function submit(event) {
    event.preventDefault();
    const { email, password } = event.target;

    if (action === 'Log In') {
      const response = await getLogIn(email.value, password.value);
      if (response) {
        dispatch(addRedirect('/index/'));
      }
    } else if (getSignUp(email.value, password.value)) {
      return <Redirect to="/index/" />;
    }
    return false;
  }

  if (redirect) {
    return <Redirect to={redirect} />;
  }
  return (
    <form className="user-form" onSubmit={(event) => submit(event)}>
      <section className="email">
        <label htmlFor="email">
          Email
          <input id="email" type="text" />
        </label>
      </section>
      <section className="password">
        <label htmlFor="password">
          Password
          <input id="password" type="password" />
        </label>
      </section>
      <button className="submit" type="submit">
        {action}
      </button>
    </form>
  );
};

UserForm.propTypes = {
  action: PropTypes.string.isRequired,
};

export default UserForm;
