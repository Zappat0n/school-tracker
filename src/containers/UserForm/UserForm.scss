@import "../../index.scss";

.user-form {
  @include responsive;

  display: flex;
  flex-direction: column;
  padding: 1rem;
  font: normal normal 700 1.5rem/2.5rem 'Free Sans', sans;

  label {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    color: $primary-blue;
    input {
      border: 1px solid $primary-blue;
      border-radius: 0.25rem;
      color: $primary-blue;
      font: normal normal 700 1rem/1.5rem 'Free Sans', sans;
    }
  }

  .submit {
    background: $primary-blue;
    margin-top: 1rem !important;
    border-radius: 0.5rem;
    width: 50%;
    margin: 0 auto;
    border: none;
    outline:none;
    font-size: 1.5rem;
    color: white;
  }

  .submit:hover {
    cursor: pointer;
  }
}
