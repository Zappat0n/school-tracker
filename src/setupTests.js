// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import { createStore, combineReducers } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import userReducer from './slices/userSlice';
import classroomTableReducer from './slices/classroomTableSlice';

const rootReducer = combineReducers({
  user: userReducer, classroomTable: classroomTableReducer,
});

const initialState = {
  user: {
    tables: {},
    title: '',
  },
  classroomTable: {
    students: [],
    presentations: [],
    scores: {},
  },
};

const store = createStore(rootReducer, initialState);

const renderComponent = (component) => render(
  <Provider store={store}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>,
);

const rendererForComponent = (component) => renderer.create(
  <Provider store={store}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>,
);

export { renderComponent, rendererForComponent, store };
