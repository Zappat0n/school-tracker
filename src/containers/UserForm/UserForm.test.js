import React from 'react';
import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import { renderComponent, rendererForComponent, store } from '../../setupTests';
import UserForm from './UserForm';

beforeAll(() => renderComponent(
  <>
    <UserForm action="Log In" />
  </>,
));

test('Display title', () => {
  expect(store.getState().user.title).toBe('Log In');
});

test('Renders Input fields correctly', () => {
  renderComponent(
    <>
      <UserForm action="Log In" />
    </>,
  );

  const inputEmail = screen.getByLabelText('Email');
  expect(inputEmail).toBeInTheDocument();
  expect(inputEmail).toHaveAttribute('type', 'text');

  const inputPassword = screen.getByLabelText('Password');
  expect(inputPassword).toBeInTheDocument();
  expect(inputPassword).toHaveAttribute('type', 'password');
});

test('Stock Match SnapShot', () => {
  const tree = rendererForComponent(
    <>
      <UserForm action="Log In" />
    </>,
  );
  expect(tree).toMatchSnapshot();
});
