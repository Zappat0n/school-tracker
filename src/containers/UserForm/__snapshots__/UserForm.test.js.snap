// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stock Match SnapShot 1`] = `
<form
  className="user-form table"
  onSubmit={[Function]}
>
  <section
    className="email"
  >
    <label
      htmlFor="email"
    >
      Email
      <input
        id="email"
        type="text"
      />
    </label>
  </section>
  <section
    className="password"
  >
    <label
      htmlFor="password"
    >
      Password
      <input
        autoComplete="on"
        id="password"
        type="password"
      />
    </label>
  </section>
  <section
    className="buttons"
  >
    <button
      className="submit"
      type="submit"
    >
      Log In
    </button>
    <div
      className="link-container"
    >
      <a
        className="link"
        href="/user/signup"
        onClick={[Function]}
      >
        Sign up
      </a>
    </div>
  </section>
</form>
`;
